client/src/components/Chat.jsx

```diff
@@ -96,25 +100,53 @@ const Chat = () => {
 	};
 
 	const sortedMessages = useMemo(
-		() =>
-			Object.values(state[app.currentChatId]?.messages ?? {})?.sort((a, b) => {
+		 () =>
+			  Object.values(state[app.currentChatId]?.messages ?? {})?.sort((a, b) => {
 				const da = new Date(a.time),
 					db = new Date(b.time);
 				return da - db;
-			}),
+			}) 
+		,
 		[state, app.currentChatId]
-	);
+	);	
+	  
+	const arrayBufferToBase64 = (arrayBuffer) => {
+		let binary = '';
+		const bytes = new Uint8Array(arrayBuffer);
+		const len = bytes.byteLength;
+		for (let i = 0; i < len; i++) {
+		  binary += String.fromCharCode(bytes[i]);
+		}
+		return btoa(binary);
+	};
```

Comment:
since this doesn't directly change state, so put this function in chatHelper, then call it